ethportal-api/src/types/consensus/kzg_commitment.rs

```diff
@@ -0,0 +1,138 @@
+use c_kzg::BYTES_PER_COMMITMENT;
+use ethereum_hashing::hash_fixed;
+use serde::{
+    de::{Deserialize, Deserializer},
+    ser::{Serialize, Serializer},
+};
+use ssz_derive::{Decode, Encode};
+use std::{
+    fmt,
+    fmt::{Debug, Display, Formatter},
+    str::FromStr,
+};
+use tree_hash::{Hash256, PackedEncoding, TreeHash};
+
+pub const VERSIONED_HASH_VERSION_KZG: u8 = 0x01;
+
+#[derive(Clone, Copy, Encode, Decode, PartialEq, Eq, Hash)]
+#[ssz(struct_behaviour = "transparent")]
+pub struct KzgCommitment(pub [u8; BYTES_PER_COMMITMENT]);
+
+impl KzgCommitment {
+    pub fn calculate_versioned_hash(&self) -> Hash256 {
+        let mut versioned_hash = hash_fixed(&self.0);
+        versioned_hash[0] = VERSIONED_HASH_VERSION_KZG;
+        Hash256::from_slice(versioned_hash.as_slice())
+    }
+
+    pub fn empty_for_testing() -> Self {
+        KzgCommitment([0; c_kzg::BYTES_PER_COMMITMENT])
+    }
+}
+
+impl From<KzgCommitment> for c_kzg::Bytes48 {
+    fn from(value: KzgCommitment) -> Self {
+        value.0.into()
+    }
+}
+
+impl Display for KzgCommitment {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "0x")?;
+        for i in &self.0[0..2] {
+            write!(f, "{i:02x}")?;
+        }
+        write!(f, "…")?;
+        for i in &self.0[BYTES_PER_COMMITMENT - 2..BYTES_PER_COMMITMENT] {
+            write!(f, "{i:02x}")?;
+        }
+        Ok(())
+    }
+}
+
+impl TreeHash for KzgCommitment {
```

Comment:
It seems worth adding tests specifically for `tree_hash_root()` if we're custom implementing this trait